src/event_rule.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+import constants
+
+""" Module for defining a rule to classify log lines into events """
+
+class EventRule:
+    """ A rule that can be applied to a line of the log to check for an event
+    
+        Attributes
+        ----------
+        tag: A string name for the event
+        regular_expression: A string holding the regular expression for
+                            a specific event
+        message_type [optional]: An int enum for the type of message the
+                                 log produces for this event 
+    """
+
+    def __init__(self, tag, regular_expression, message_type=None):
+        allowed_message_types = [constants.MessageType.INFO,
```

Comment:
Is there any reason to have allowed message types be defined separately from all of the enum values?

Reply:
Good point! I found a better way to check against all possible message type enums.
"allowed_message_types = [message_type.value for message_type in constants.MessageType]"
This should be more maintainable since it doesn't directly call out each enum.